forum/qa-plugin/users-hidden-posts/qa-users-hidden-posts.php

```diff
@@ -0,0 +1,144 @@
+                    'tags' => 'name=comments',
+                    'value' => isset($_POST['comments']),
+                ],
+                'answers' => [
+                    'type' => 'checkbox',
+                    'label' => qa_lang_html('users-hidden-posts/answers'),
+                    'tags' => 'name=answers',
+                    'value' => isset($_POST['answers']) ? true : false,
+
+                ],
+                'questions' => [
+                    'type' => 'checkbox',
+                    'label' => qa_lang_html('users-hidden-posts/questions'),
+                    'tags' => 'name=questions',
+                    'value' => isset($_POST['questions']) ? true : false,
+                ],
+            ],
+
+            'buttons' => [
+                'submit' => [
+                    'tags' => 'name="search"',
+                    'label' => qa_lang_html('users-hidden-posts/search-label'),
+                ],
+            ],
+        ];
+
+        $qa_content['custom'] = $this->showUsersHiddenPosts();
+        return $qa_content;
+    }
+
+    private function getUsersHiddenPosts()
+    {
+        if(isset($_POST['search'])) {
+            $comments = isset($_POST['comments']) ? 'C': false;
+            $answers = isset($_POST['answers']) ? 'A' : false;
+            $questions = isset($_POST['questions']) ? 'Q' : false;
+            $userID = qa_handle_to_userid($this->userHandle);
+            $sufix = "_HIDDEN";
+
+            if((!$comments && !$answers && !$questions) || !is_numeric($userID)) {
+                return false;
+            }
+            
+            $sql = "SELECT org.title, org.parentid, org.postid, org.created, org.type, par.title AS parentTitle FROM ^posts as org 
+                    LEFT JOIN ^posts as par ON org.parentid=par.postid WHERE org.userid = $ AND (org.type =";
+
+            if($comments) {
+                $sql .=  "'".$comments.$sufix."'";
+            }
+
+            
+            if($answers) {
+                $sql .= $comments ? ' OR org.type=':'';
+                $sql .= "'".$answers.$sufix."'";
+            }
+
+            if($questions) {
+                $sql .= ($comments || $answers) ? ' OR org.type=':'';
+                $sql .= "'".$questions.$sufix."'";
+            }
+
+            $sql .= ") ORDER BY org.postid DESC"; 
+            $results = qa_db_read_all_assoc(qa_db_query_sub($sql, $userID));
+
+            return $results ? $results : false;
+        }
+    }
+
+    private function showUsersHiddenPosts()
+    {
+        $hidden = $this->getUsersHiddenPosts();
+        $postsHtml ="";
+
+        if(!$hidden) {
+            return qa_lang_html("users-hidden-posts/no-results");
+        }
+        
+
+        foreach($hidden as $post) {
+            $title = $post['parentTitle'] ?? $post['title'];
+            $id = !empty($post['parentid']) ? $post['parentid'] : $post['postid'];
+
+            $postsHtml .= '<div class = "post">
```

Comment:
1. Zamieniłbym tę div-ową strukturę na uporządkowaną listę (bo AFAIK kolejność nadaje chronologia dat). Na przykładzie:
```html
<ol>
    <li>
        Komentarz do:
        <a class="postsHeader" href="../6/pytanie-eksperta-kategorii-css-i-html?show=12#c12">Pytanie eksperta kategorii CSS i HTML</a>
        <p>Dodano: 2023-02-14 </p>
    </li>
    <li>
        Odpowiedź na:
        <a class="postsHeader" href="../6/pytanie-eksperta-kategorii-css-i-html?show=11#a11">Pytanie eksperta kategorii CSS i HTML</a>
        <p>Dodano: 2023-02-14 </p>
    </li>
</ol>
```
Dodatkowo, same daty można wrzucić do znacznika [`<time>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time):
```html
<p>Dodano: <time>2023-02-14</time></p>
```

2. Z tego co widzę, CSS-owe klasy **post** i **postsHeader** nie są używane, więc raczej do usunięcia

Reply:
>(bo AFAIK kolejność nadaje chronologia dat)

Akurat tutaj mamy odwrotną chronologię. Zaczynamy od najnowszych, na najstarszych kończymy

Reply:
Ok, ale nadal kolejność jest istotna. :P Na upartego można użyć [atrybutu `reversed`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol#attr-reversed), żeby lista numerowała w odwrotnej kolejności.

Reply:
Niby można, ale trochę to może być mylące.
Na przykład

 Mamy 100 wyników i sobie idzie od 100 do 1

Wg mnie to trochę nie ma sensu, taka odwrotna numeracja